### DISCUSSION TITLE
How to set a query correctly

### DISCUSSION THREAD

**Discussion by lewysigns** [anvilistas/tabulator#76] [Q&A] [answered]
I am looking at the README which state the following:

**It is also possible to set queries on the data, similar to JS Tabulator's set_filter() method. Instead use set_query(). Any valid query using Anvil's query API will be supported.**

```
import anvil.tables.query as q

def text_box_1_change(self, **event_args):
    name = self.text_box_1.text
    if not name:
        self.tabulator.clear_query()
    else:
        self.tabulator.set_query(name=q.ilike(f"%{name}%"))

```

However when I attempt to set my query I get the following error:

```
TypeError: __call__() takes no keyword arguments
at Tabulator_copy, line 39
```

Here is the Code:

```
def button_query_click(self, **event_args):
  """This method is called when the button is clicked"""
  self.tabulator.set_query(row_num=3)
```
Here is a link to a simple example. (just click the "SET QUERY" button.

https://anvil.works/build#clone:TX5IJADW5BLXJTRB=IGU6N4GJMVJGZXZHMS4NRCBH

**Comment by s-cork** [comment, marked as answer]
Let's move this to an issue. It seems to be a regression in 2.2.x. Works as expected in 2.1.6.

**Reply by s-cork** [reply]
This should be fixed now on published/development and v2.2.2
The tabulator library changed something internally which affected the javascript python dance here

**Reply by lewysigns** [reply]
You work to quick for me to react 👍🏽 

tysm Stu!
